### PR TITLE
Performance tests

### PR DESCRIPTION
**Pull request by adevnadia**
Missing benchmark is returned;
Karma supports now unit tests and performance tests;

### CODE REVIEW

File: sauce.browsers.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
+const sauceBrowsers = {
```
Comment (treshugart) [inline sauce.browsers.js:1]: What's the reasoning for moving this out?

**Comment by treshugart** [comment]
lgtm

File: sauce.browsers.js (hunk 1 above)
Comment (adevnadia) [inline sauce.browsers.js:1]: We have webpack.config in a separate file, why not for the saucelab?
Plus, it's really long, and I think it just make sense to put this in a separate file.

Comment (treshugart) [inline sauce.browsers.js:1]: Ok, cool. Just wondering if it was anything else. Did you consider moving it into a folder? The only thing I can think of is a possible future where we start moving more stuff out, but maybe that's premature optimisation.

File: karma.conf.js [hunk 2]
```diff
@@ -1,144 +1,19 @@
...
+const sauceBrowsers = require('./sauce.browsers');
 
 module.exports = function (config) {
+  var files = ['test/unit.js'];
+  var preprocessors = {
+    'test/unit.js': [ 'webpack', 'sourcemap' ]
+  };
+
+  if (process.argv.indexOf('--perf') > -1) {
+    files = [ 'node_modules/benchmark/benchmark.js', 'test/perf.js' ];
```
Comment (joscha) [inline karma.conf.js]: maybe use `require.resolve` here?

File: karma.conf.js [hunk 3]
```diff
@@ -1,144 +1,19 @@
...
+    preprocessors = {
+      'test/perf.js': [ 'webpack', 'sourcemap' ]
```
Comment (joscha) [inline karma.conf.js:19]: do we need source maps for perf?

File: karma.conf.js [hunk 4]
```diff
@@ -150,17 +25,15 @@ module.exports = function (config) {
 
     // list of files / patterns to load in the browser
     // all dependancies should be traced through here
-    files: [ 'test/unit.js' ],
+    files: files,
 
     // list of files to exclude
     exclude: [],
 
     // preprocess matching files before serving them to the browser
```
Comment (joscha) [inline karma.conf.js]: either move this comment with the block or remove it (I think it can be removed)

Comment (adevnadia) [inline karma.conf.js]: Why, what is wrong with it?

File: karma.conf.js (hunk 3 above)
Comment (treshugart) [inline karma.conf.js:19]: I don't think so, but it might be nice.

File: sauce.browsers.js (hunk 1 above)
Comment (jpnelson) [inline sauce.browsers.js:1]: I think the file name could be `sauce-browsers.js`

File: karma.conf.js (hunk 4 above)
Comment (jpnelson) [inline karma.conf.js]: The comment was relating to the specific preprocessors, so it'd make sense to move it with the code block – that info could easily be found by looking at the karma site though

Comment (adevnadia) [inline karma.conf.js]: updated here, is this what you mean?

File: sauce.browsers.js (hunk 1 above)
Comment (adevnadia) [inline sauce.browsers.js:1]: I checked just for fun, and turned out that `sauce.browsers` is more common name than `sauce-browsers` :-)
In `chai` they are using `sauce.browsers`, and no one using `sauce-browsers`.

File: karma.conf.js (hunk 2 above)
Comment (adevnadia) [inline karma.conf.js]: done

**Comment by joscha** [comment]
LGTM

**Pull request merged**